commands/marketplace_publish.go

```diff
@@ -48,15 +47,19 @@ func (c *marketplacePublishCmd) preRunE(cmd *cobra.Command, args []string) error
 	c.path = getFirstOrCurrentPath(args)
 
 	var err error
-	c.sid, c.hash, err = deployService(c.e, c.path, map[string]string{})
+	_, hash, err := deployService(c.e, c.path, map[string]string{})
 	if err != nil {
 		return err
 	}
-	fmt.Printf("%s Service deployed with sid %s and hash %s\n", pretty.SuccessSign, pretty.Success(c.sid), pretty.Success(c.hash))
+	c.service, err = c.e.ServiceByID(hash)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%s Service deployed with sid %s and hash %s\n", pretty.SuccessSign, pretty.Success(c.service.Definition.Sid), pretty.Success(c.service.Definition.Hash))
```

Comment:
it might be better to explicitly use `hash` got from `deployService()` here. because in future, we'll make service sids to have one-to-many relationships with service hashes.  

thus, `ServiceByID()` may return multiple versions and maybe in desc order on deploy time but it's not guaranteed that the last hash in the array will be belong to the last publish because two publish commands may run in parallel for the different versions of same service.

Reply:
if `ServiceByID(**sid**)` is going to return the last service version and `ServiceByID(**hash**)` will return the same version with `hash`, then all ok.

Reply:
anyway, we're also using `c.service.Definition` so we have to make this call. let's make sure that `ServiceByID(hash)` will return the version for `hash` in future and `ServiceByID(sid)` may return the latest hash.